Allow users to download their creation!

mogest opened the pull request:
I've put in a little download button and it turns the user's creation into a PNG and lets them download it.

It's a bit hacky as you can't (easily) render html into a canvas, so I have to manually render the middle box and text.  Means if you change the html then it won't match the downloaded file :(  But good enough for now?

Commit:
Allow users to download their creation!

mogest commented:
Oh I missed the background gradient :(

Commit:
Downloaded file now with background gradient

mogest commented:
Better.

mermop commented on public/script.js:
```diff
@@ -6,6 +6,38 @@ $( document ).ready(function() {
+    outputContext.font = "bold 24pt Times";
+
+    var gradient = outputContext.createLinearGradient(0, 0, canvas.width, 0);
+    gradient.addColorStop(0, "#ECF9FF");
```
Hmm, so the initial goal was to change the background gradient on a daily basis. I never ended up automating that but the colours are now defined in index.js as `background_colours` - it'd be good to avoid having to define those colours twice. 

Additionally, I've been changing the angle of the background gradient as well as the colours - this won't be able to do that, will it? 

Perhaps it's possible to have the gradient background on the canvas all along?

mermop replied:
seems to work fine though
![advice-2](https://cloud.githubusercontent.com/assets/1615322/8848703/930614fc-318d-11e5-855e-281eb5a2acd1.png)

mogest replied:
The gradient being on the canvas would hide the text :(
Is OK, I'll get the gradient info from the variables. Yep we can do angles too, I'll have to dust off my sixth form trigonometry! Will do so in next couple of days.
On 23 Jul 2015 10:46 pm, Merrin Macleod notifications@github.com wrote:In public/script.js:

> @@ -6,6 +6,38 @@ $( document ).ready(function() {
>    canvas.height = window.innerHeight;
>    drawStuff();
> -  $("#download").click(function() {
> -    var outputCanvas = document.createElement("canvas");
> -    outputCanvas.width = canvas.width;
> -    outputCanvas.height = canvas.height;
>   +
> -    var outputContext = outputCanvas.getContext("2d");
> -    outputContext.font = "bold 24pt Times";
>   +
> -    var gradient = outputContext.createLinearGradient(0, 0, canvas.width, 0);
> -    gradient.addColorStop(0, "#ECF9FF");

Hmm, so the initial goal was to change the background gradient on a daily basis. I never ended up automating that but the colours are now defined in index.js as background_colours - it'd be good to avoid having to define those colours twice. 

Additionally, I've been changing the angle of the background gradient as well as the colours - this won't be able to do that, will it? 

Perhaps it's possible to have the gradient background on the canvas all along? 

—Reply to this email directly or view it on GitHub.

mermop replied:
awesome!

![advice 3](https://cloud.githubusercontent.com/assets/1615322/8849600/120593fc-3195-11e5-87f2-f4c593d14e35.png)

Commit:
Use gradient angle + colours defined in index.js

Now we use the gradient details defined in the backend and passed through
to the front end, instead of having it hardcoded in two places and
not supporting gradient angles.  Also took the opportunity to refactor
a bit so the download click event isn't so huge.

mogest commented:
Hopefully this does what you're after :rainbow:

mermop commented:
✨🎈Yay thank you 🌈🎈✨
